src/flow_node_instance_service.ts

```diff
@@ -20,4 +35,144 @@ export class FlowNodeInstanceService implements APIs.IFlowNodeInstanceManagement
     return this.flowNodeInstanceService.queryByProcessInstance(processInstanceId, offset, limit);
   }
 
+  public async getAllSuspendedTasks(
+    identity: IIdentity,
+    offset: number = 0,
+    limit: number = 0,
+  ): Promise<DataModels.FlowNodeInstances.TaskList> {
+
+    const suspendedFlowNodes = await this.flowNodeInstanceService.queryByState(FlowNodeInstanceState.suspended);
+
+    const emptyActivityList = await this.emptyActivityConverter.convert(identity, suspendedFlowNodes);
+    const manualTaskList = await this.manualTaskConverter.convert(identity, suspendedFlowNodes);
+    const userTaskList = await this.userTaskConverter.convert(identity, suspendedFlowNodes);
+
+    const totalCount = emptyActivityList.emptyActivities.length + manualTaskList.manualTasks.length + userTaskList.userTasks.length;
+
+    const tasks = emptyActivityList.emptyActivities.concat(manualTaskList.manualTasks, userTaskList.userTasks);
```

Comment:
Array.concat ist ziemlich langsam.

Ich würde hier eher `Array.push` mit Spread Operatoren verwenden.

```suggestion
    const tasks = Array.prototype.push(...emptyActivityList.emptyActivities, ...manualTaskList.manualTasks, ...userTaskList.userTasks);
```

Siehe:
https://dev.to/uilicious/javascript-array-push-is-945x-faster-than-array-concat-1oki

Reply:
1. push returned leider nur die Länge des Arrays nach dem `push`
2. Aus dem Artikel:
""In conclusion, it is true that the main reason why concat is so much slower than .push is simply that it creates a new array and does the additional work to copy the first array over."

Sprich wir könnten lediglich die anderen Tasks in ein bestehendes Array reinfügen und das Array dann benutzen (was funktioniert, aber unschön ist) oder wenn wir die Tasks in ein neues Array pushen hätten wir scheinbar die gleiche Performance wie mit concat.

Reply:
Mein Fehler. Das muss natürlich heißen:

```suggestion
    const tasks = [];
   tasks.push(...emptyActivityList.emptyActivities, ...manualTaskList.manualTasks, ...userTaskList.userTasks);
```

Ich pass das in den anderen Suggestions an.

>  wenn wir die Tasks in ein neues Array pushen hätten wir scheinbar die gleiche Performance wie mit concat.

Nope. Der Performanceverlust ist zwar nicht so dramatisch wie bei dem anderen dargestellten UseCase (50000 kleine Arrays mergen), aber immernoch ziemlich hoch (90% Unterschied ist schon ne Menge).
Steht da auch so im Artikel.

Ich denke, dass das schon einen Unterschied machen kann, sobald man die 1000er Grenze an FlowNodeInstanzen überschritten hat - was ziemlich schnell geht. Die Steag z.B. ist da schon lange drüber.

Reply:
Mag sein dass ich blind bin, aber den Fall hab ich in dem Artikel nicht gefunden 🤔 .

Hab das ganze mal fix nachgebaut und dabei scheint concat sogar schneller zu sein:

https://jsperf.com/sk-concat-vs-push/

<img width="981" src="https://user-images.githubusercontent.com/20394992/65129117-fa8b5f00-d9fa-11e9-8bce-91d201380064.png">

Reply:
Ok, um heute mit dem Release vorwärts zu kommen, lass das dann erstmal so.
Ich schau mir das dann nochmal näher an, wenn die beta raus ist.